web/wp-content/themes/dctx-v2/inc/acf.php

```diff
@@ -0,0 +1,449 @@
+}
+
+/**
+ * Load colors dynamically into select field from dctx_get_theme_colors()
+ *
+ * @author WDS
+ * @param array $field field options.
+ * @return array new field choices.
+ *
+ * @author Corey Colins <corey@webdevstudios.com>
+ */
+function dctx_acf_load_color_picker_field_choices( $field ) {
+
+	// Reset choices.
+	$field['choices'] = array();
+
+	// Grab our colors array.
+	$colors = dctx_get_theme_colors();
+
+	// Loop through colors.
+	foreach ( $colors as $key => $color ) {
+
+		// Create display markup.
+		$color_output = '<div style="display: flex; align-items: center;"><span style="background-color:' . esc_attr( $color ) . '; border: 1px solid #ccc;display:inline-block; height: 15px; margin-right: 10px; width: 15px;"></span>' . esc_html( $key ) . '</div>';
+
+		// Set values.
+		$field['choices'][ sanitize_title( $key ) ] = $color_output;
+	}
+
+	// Return the field.
+	return $field;
+}
+add_filter( 'acf/load_field/name=color_picker', 'dctx_acf_load_color_picker_field_choices' );
+
+/**
+ * Get the theme colors for this project. Set these first in the Sass partial then migrate them over here.
+ *
+ * @author WDS
+ * @return array The array of our color names and hex values.
+ * @author Corey Collins
+ */
+function dctx_get_theme_colors() {
+	return array(
+		esc_html__( 'Alto', 'dctx' )           => '#ddd',
+		esc_html__( 'Black', 'dctx' )          => '#000',
+		esc_html__( 'Blue', 'dctx' )           => '#21759b',
+		esc_html__( 'Cod Gray', 'dctx' )       => '#111',
+		esc_html__( 'Dove Gray', 'dctx' )      => '#666',
+		esc_html__( 'Gallery', 'dctx' )        => '#eee',
+		esc_html__( 'Gray', 'dctx' )           => '#808080',
+		esc_html__( 'Gray Alt', 'dctx' )       => '#929292',
+		esc_html__( 'Light Yellow', 'dctx' )   => '#fff9c0',
+		esc_html__( 'Mineshaft', 'dctx' )      => '#333',
+		esc_html__( 'Silver', 'dctx' )         => '#ccc',
+		esc_html__( 'Silver Chalice', 'dctx' ) => '#aaa',
+		esc_html__( 'White', 'dctx' )          => '#fff',
+		esc_html__( 'Whitesmoke', 'dctx' )     => '#f1f1f1',
+	);
+}
+
+/**
+ * Adds h1 or h2 heading for hero based on location.
+ *
+ * @author WDS
+ * @param string $block_title acf value.
+ * @author jomurgel <jo@webdevstudios.com>
+ * @return void
+ */
+function dctx_display_hero_heading( $block_title ) {
+
+	// Bail if our title is empty.
+	if ( empty( $block_title ) ) {
+		return;
+	}
+
+	// Set hero title to h1 if it's the first block not on the homepage.
+	$index   = get_row_index();
+	$heading = 1 === $index && ! ( is_front_page() && is_home() ) ? 'h1' : 'h2';
+
+	echo sprintf( '<%1$s class="hero-title">%2$s</%1$s>', esc_attr( $heading ), esc_html( $block_title ) );
+}
+
+/**
+ * Echo link function
+ *
+ * @param array $args defaults args - link array and whether or not to append button class.
+ * @author jomurgel <jo@webdevstudios.com>
+ * @since NEXT
+ */
+function dctx_display_link( $args = array() ) {
+	echo dctx_get_link( $args ); // WPCS: XSS Ok.
```

Comment:
:warning: **Warning**: Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native &quot;phpcs:ignore Standard.Category.SniffName.ErrorCode&quot; annotations instead. Found: // WPCS: XSS Ok.\n (*WordPress.Security.EscapeOutput.DeprecatedWhitelistCommentFound*).